src/NHibernate/Engine/Cascade.cs

```diff
@@ -203,7 +203,7 @@ private void CascadeProperty(object parent, object child, IType type, CascadeSty
 
 							if (loadedValue != null)
 							{
-								eventSource.Delete(entry.Persister.EntityName, loadedValue, false, null);
```

Comment:
This `entry` is for parent. So parent entity name is supplied for removing child entity. This entity name is only used when one-to-one entity is detached - so it's quite rare case.